### PR TITLE
CCv0: Refactor ccv0.sh to remove duplicate code

### PR DESCRIPTION
**Pull request by GeorginaKin**
Refactored ccv0.sh to remove rootfs code now in lib.sh

Fixes: #4512

Co-authored-by: Megan Wright Megan.Wright@ibm.com
Signed-off-by: Georgina Kinge <georgina.kinge@ibm.com>

### CODE REVIEW

**Comment by GeorginaKin** [comment]
Hey again @arronwy I please have another re-run of the TDX job here 😄 Thank you so much!!

**Comment by stevenhorsman** [comment]
@GeorginaKin  - did you get this to work manually? I'm hitting problems:
```
~/ccv0.sh kubernetes_create_ssh_demo_pod
...
service/ccv0-ssh created
deployment.apps/ccv0-ssh created
error: timed out waiting for the condition on pods/ccv0-ssh-9fd8559d8-m77xz
[ccv0.sh:118] ERROR: kubectl wait --for=condition=ready pods/ccv0-ssh-9fd8559d8-m77xz

$ kubectl get pods
NAME                       READY   STATUS              RESTARTS   AGE
ccv0-ssh-9fd8559d8-m77xz   0/1     ContainerCreating   0          40s
stevenh@ccv0-ssh-demo-k8s1:~$ kubectl describe pods
Name:           ccv0-ssh-9fd8559d8-m77xz
Namespace:      default
...
Events:
  Type     Reason                  Age                From               Message
  ----     ------                  ----               ----               -------
  Normal   Scheduled               46s                default-scheduler  Successfully assigned default/ccv0-ssh-9fd8559d8-m77xz to ccv0-ssh-demo-k8s1.fyre.ibm.com
  Warning  FailedCreatePodSandBox  18s (x2 over 37s)  kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create containerd task: failed to create shim task: Failed to Check if grpc server is working: ttrpc: closed: unknown
 ```

**Comment by stevenhorsman** [comment]
I think I've found the problem.
The test code is copying the agent-config to a `tests/fixtures` directory in the rootfs,  `"agent.config_file=/tests/fixtures/$(basename ${rootfs_agent_config})"` (based on code that Wainer added)
whereas `ccv0.sh` is looking for it in a different place:
```
add_kernel_params "agent.config_file=/etc/agent-config.toml"
```
I think we need to review if the code in `ccv0.sh` that sets `add_kernel_params "agent.config_file=/etc/agent-config.toml"` is still required any more and give it a manual test before pushing the PR

**Comment by arronwy** [comment]
> Hey again @arronwy Can I please have another re-run of the TDX job here 😄 Thank you so much!!

Done ：）

File: docs/how-to/ccv0.sh [hunk 1]
```diff
@@ -622,8 +608,8 @@ main() {
         agent_create_container)
             agent_create_container
             ;;
-        copy_files_to_guest)
-            call_copy_files_to_guest
+        copy_signature_files_to_guest)
```
Comment (stevenhorsman) [inline docs/how-to/ccv0.sh:611]: Can you update the how-to doc to reflect this name change please.

**Review by stevenhorsman — CHANGES_REQUESTED** [review]

**Comment by stevenhorsman** [comment]
The manual ssh-demo tests worked:
```
~/ccv0.sh kubernetes_create_ssh_demo_pod
...
service/ccv0-ssh created
deployment.apps/ccv0-ssh created
pod/ccv0-ssh-9fd8559d8-lnf2f condition met
NAME                       READY   STATUS    RESTARTS   AGE
ccv0-ssh-9fd8559d8-lnf2f   1/1     Running   0          16s
stevenh@ssh-demo-21:~$ ~/ccv0.sh connect_to_ssh_demo_pod
Issuing command 'ssh -i /home/stevenh/go/src/github.com/confidential-containers/documentation/demos/ssh-demo/ccv0-ssh root@10.99.51.218'
The authenticity of host '10.99.51.218 (10.99.51.218)' can't be established.
ED25519 key fingerprint is SHA256:wK7uOpqpYQczcgV00fGCh+X97sJL3f6G1Ku4rvlwtR0.
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
Warning: Permanently added '10.99.51.218' (ED25519) to the list of known hosts.
Welcome to Alpine!

The Alpine Wiki contains a large amount of how-to guides and general
information about administrating Alpine systems.
See <http://wiki.alpinelinux.org/>.

You can setup the system with the command: setup-alpine

You may change this message by editing /etc/motd.

ccv0-ssh-9fd8559d8-lnf2f:~# exit
```

**Comment by stevenhorsman** [comment]
The signature-verification tests fail:
```
~/ccv0.sh copy_signature_files_to_guest
kata-containers/
kata-containers/confidential-containers/
kata-containers/confidential-containers@sha256=febcf61cd6e1ac9628f6ac14fa40836d16f3c6ddef3b303ff0321606e55ddd0b/
kata-containers/confidential-containers@sha256=febcf61cd6e1ac9628f6ac14fa40836d16f3c6ddef3b303ff0321606e55ddd0b/signature-1
kata-containers/confidential-containers@sha256=7250923ba3543110040462388756ef099331822c6172a050b12c7a38361ea46f/
kata-containers/confidential-containers@sha256=7250923ba3543110040462388756ef099331822c6172a050b12c7a38361ea46f/signature-1
stevenh@ccv0-sig-ver1:~$ ~/ccv0.sh crictl_create_cc_pod
Pod config file '/home/stevenh/go/src/github.com/kata-containers/tests/integration/containerd/confidential/fixtures/pod-config.yaml' does not exist
[ccv0.sh:399] ERROR: return 1
```

@GeorginaKin - I think you might have forgotten to rebase your changes on the latest `CCv0` branch?

**Comment by stevenhorsman** [comment]
`crictl` (without `skopeo`) tests are also working:
```
$ ~/ccv0.sh crictl_create_cc_pod
# Copyright (c) 2021, 2022 IBM Corporation
#
# SPDX-License-Identifier: Apache-2.0
#
metadata:
  #  name:
  name: kata-cc-busybox-sandbox
  uid: 6295f730-df85-4949-ad22-28ade940a419
  namespace: default
dns_config:
  servers:
    - 8.8.8.8
POD ID              CREATED             STATE               NAME                      NAMESPACE           ATTEMPT             RUNTIME
75fab8d59530f       6 seconds ago       Ready               kata-cc-busybox-sandbox   default             0                   kata
stevenh@stevenh-ccv0-not-skopeo1:~$ ~/ccv0.sh crictl_create_cc_container
0eb29a7d76710cb1223e4a960f83e52ced94b62c116a203da169bdc9762e6b12
CONTAINER           IMAGE                                                    CREATED                  STATE               NAME                     ATTEMPT             POD ID              POD
0eb29a7d76710       quay.io/kata-containers/confidential-containers:signed   Less than a second ago   Running             kata-cc-busybox-signed   0                   75fab8d59530f       unknown
stevenh@stevenh-ccv0-not-skopeo1:~$ ~/ccv0.sh crictl_delete_cc
0eb29a7d76710cb1223e4a960f83e52ced94b62c116a203da169bdc9762e6b12
0eb29a7d76710cb1223e4a960f83e52ced94b62c116a203da169bdc9762e6b12
Stopped sandbox 75fab8d59530f7216baa2125c2d69c5747b533709e2ed624c68c27bf1921e05c
Removed sandbox 75fab8d59530f7216baa2125c2d69c5747b533709e2ed624c68c27bf1921e05c
```

**Comment by stevenhorsman** [comment]
It looks like the fedora build is timing out on building the agent, which anecdotally has been getting slower and slower recently. Once last try of /test-fedora 🤞

**Review by stevenhorsman — APPROVED** [review]
The code and behaviour looks good to me!

**Pull request merged**